### PR TITLE
fix: organization not getting displayed after login

### PR DESCRIPTION
**Pull request by bjp232004**
## Summary by CodeRabbit

- **New Features**
	- Enhanced organization management with improved responsiveness to user interactions.
	- Added a watcher to track changes in the selected organization, ensuring prompt updates to the layout.
- **Bug Fixes**
	- Improved error handling and logging during organization selection.
- **Refactor**
	- Restructured organization change handling for better performance and clarity.
	- Updated logic for setting the default organization when none is selected.

### CODE REVIEW

File: web/src/layouts/MainLayout.vue [hunk 1]
```diff
@@ -698,6 +690,10 @@ export default defineComponent({
       );
     };
 
+    watch(store.state.selectedOrganization, () => {
+      selectedOrg.value = store.state.selectedOrganization;
+    });
+
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/layouts/MainLayout.vue]: Removed this code

File: web/src/layouts/MainLayout.vue [hunk 2]
```diff
@@ -1057,10 +1076,12 @@ export default defineComponent({
...
     },
-    changeOrganization() {
-      setTimeout(() => {
+    changeOrganization: {
+      handler() {
         this.setSelectedOrganization();
-      }, 500);
+      },
+      deep: true,
+      immediate: true,
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/layouts/MainLayout.vue:1079]: We want to do deep check

File: web/src/layouts/MainLayout.vue [hunk 3]
```diff
@@ -698,6 +692,11 @@ export default defineComponent({
       );
     };
 
+    watch(store.state.selectedOrganization, () => {
+      console.log("org change", store.state.selectedOrganization, selectedOrg.value);
+      selectedOrg.value = store.state.selectedOrganization;
+    });
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/layouts/MainLayout.vue]: Removed this code from final version

File: web/src/layouts/MainLayout.vue [hunk 4]
```diff
@@ -638,6 +630,7 @@ export default defineComponent({
       try {
         const url = new URL(window.location.href);
         const localOrg: any = useLocalOrganization();
+        console.log("on before mount", localOrg.value, selectedOrg.value);
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/layouts/MainLayout.vue]: removed all console log which were added for debug purpose

File: web/src/layouts/MainLayout.vue [hunk 5]
```diff
@@ -917,6 +909,28 @@ export default defineComponent({
...
+            : "",
+          status: data.status,
+          note: data.hasOwnProperty("CustomerBillingObj")
+            ? data.CustomerBillingObj.note
+            : "",
+        };
+        selectedOrg.value = optiondata;
+        useLocalOrganization(optiondata);
+        store.dispatch("setSelectedOrganization", optiondata);
+      }
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/layouts/MainLayout.vue:932]: NA

File: web/src/layouts/MainLayout.vue [hunk 6]
```diff
@@ -1057,10 +1071,12 @@ export default defineComponent({
...
     },
-    changeOrganization() {
-      setTimeout(() => {
+    changeOrganization: {
+      handler() {
         this.setSelectedOrganization();
-      }, 500);
+      },
+      deep: true,
+      immediate: true,
```
[the remark this answers is not in the document]
Comment (bjp232004) [reply web/src/layouts/MainLayout.vue:1079]: Already provide comment

**Review by nikhilsaikethe — APPROVED** [review]

**Pull request merged**